### PR TITLE
chore: App-router-migration`(/apps`) fix metadata, extract components into `/module` and finish migration

### LINKED ISSUE #16347: [umbrella for appDir migration] Fix metadata for routes in App Router for `/apps` route group
## Nested Routes under `/future/apps` route group:
- [x] `apps`
- [x] `apps/[slug]/**/*`
- [x] `apps/categories/**/*`
- [x] `apps/installed/**/*`

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes #16347, #16292
- Fixes CAL-4020, CAL-4021, CAL-4019, CAL-4167, CAL-4017, CAL-4018, CAL-4040, CAL-4229

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/web/modules/apps/[slug]/[...pages]/pages-view.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+      notFound: true,
+    };
+  }
+
+  const parsedParams = paramsSchema.safeParse(params);
+  if (!parsedParams.success) {
+    return {
+      notFound: true,
+    };
+  }
```
Comment (hbjORbj) [inline apps/web/modules/apps/[slug]/[...pages]/pages-view.tsx:125]: This is a new block of code I added

File: apps/web/modules/apps/[slug]/[...pages]/pages-view.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+export default AppPage;
+
+const paramsSchema = z.object({
+  slug: z.string(),
+  pages: z.array(z.string()),
+});
+
+export async function getServerSideProps(
+  context: GetServerSidePropsContext
+): Promise<GetServerSidePropsResult<any>> {
```
Comment (hbjORbj) [inline apps/web/modules/apps/[slug]/[...pages]/pages-view.tsx:114]: This is a new block of code I added

**Review by joeauyeung — CHANGES_REQUESTED** [review]
@hbjORbj I noticed when I try and navigate to `/future/installation/[[...step]]` I'm being redirected back to the app store.

Other routes are working fine

**Comment by hbjORbj** [comment]
> @joeauyeung I noticed when I try and navigate to /future/installation/[[...step]] I'm being redirected back to the app store.

Hey Joe, I experienced the same with `/installation/[[...step]]`. Can you give me an example of URL you tested this route with? (a.k.a what [[...step]] value did you use?)

**Comment by joeauyeung** [comment]
> > @joeauyeung I noticed when I try and navigate to /future/installation/[[...step]] I'm being redirected back to the app store.
> 
> Hey Joe, I think this is ok because the experience is the same in Pages Router (both in dev/prod). Btw, what [[...step]] value did you use?

@hbjORbj here's what I'm experiencing
https://www.loom.com/share/dbb8bd50228247d987a879888f763a2d?sid=30594643-a418-4378-8478-13d4bf88d7dc

**Comment by hbjORbj** [comment]
@joeauyeung, Hey Joe, thanks for the screencast 🙏. You are right! I fixed the bug in this [commit](https://github.com/calcom/cal.com/pull/16354/commits/7469a51daebe22ed871f8d4cc4716c885d79f6f6).

**Review by joeauyeung — APPROVED** [review]
Installation page is now working. Great work @hbjORbj!

**Pull request merged**